lib/properties.js

```diff
@@ -267,22 +267,19 @@ internals.properties.prototype.parseString = function(property, joiObj) {
   const describe = joiObj.describe();
 
   if (describe.type !== 'date') {
-    property.minLength = internals.getArgByName(describe.rules, 'min');
-    property.maxLength = internals.getArgByName(describe.rules, 'max');
+    property.minLength = internals.getArgByName(describe.rules, 'min', 'limit');
+    property.maxLength = internals.getArgByName(describe.rules, 'max', 'limit');
   }
 
   // add regex
-  joiObj._tests.forEach(test => {
-    // Joi post v10
-    if (Utilities.isObject(test.arg) && test.arg.pattern) {
-      property.pattern = test.arg.pattern.source;
-    }
-    // Joi pre v10
-    /* $lab:coverage:off$ */
-    if (Utilities.isRegex(test.arg)) {
-      property.pattern = test.arg.source;
+  joiObj._rules.forEach(test => {
```

Comment:
I removed the pre/post split. These changes are so breaking that there is no value (IMO) of trying to make both schemas work. Every joi reference would need multiple options.